### ISSUE TITLE
missing wildcard example when evaluating `help(reshape)`

### ISSUE THREAD

**Issue by dvd101x** [josdejong/mathjs#2854]
When evaluating `help(reshape)` the examples do not show the use of *wlidcard* `-1`

```
Examples:
    reshape([1, 2, 3, 4, 5, 6], [2, 3])
        [[1, 2, 3], [4, 5, 6]]
    reshape([[1, 2], [3, 4]], [1, 4])
        [[1, 2, 3, 4]]
    reshape([[1, 2], [3, 4]], [4])
        [1, 2, 3, 4]
```

I see on the [docs ](https://mathjs.org/docs/reference/functions/reshape.html) that an example is shown but with different alignment

```
 math.reshape([1, 2, 3, 4, 5, 6], [2, 3])
 // returns Array  [[1, 2, 3], [4, 5, 6]]

 math.reshape([[1, 2], [3, 4]], [1, 4])
 // returns Array  [[1, 2, 3, 4]]

 math.reshape([[1, 2], [3, 4]], [4])
 // returns Array [1, 2, 3, 4]

 const x = math.matrix([1, 2, 3, 4, 5, 6, 7, 8])
 math.reshape(x, [2, 2, 2])
 // returns Matrix [[[1, 2], [3, 4]], [[5, 6], [7, 8]]]

math.reshape([1, 2, 3, 4], [-1, 2])
// returns Matrix [[1, 2], [3, 4]]
```

It seems to be related with either the order or the alignment of lines 30 and 31 in here 
https://github.com/josdejong/mathjs/blob/c51d263d9cc35f17058e2de02d0b32489e96a57d/src/function/matrix/reshape.js#L30

I don't know if this behavior is as intended but please review

**Comment by josdejong** [comment]
Thanks, I've added an example with a wildcard to the embedded docs now.

The example `math.reshape([1, 2, 3, 4], [-1, 2])` returning `[[1, 2], [3, 4]]` looks correct to me: the wild card value for the rows is computed as the total number of elements divided by the number of columns: `4 / 2 = 2`, and the result is a 2x2 matrix.

I noticed a bug in `reshape` when testing the examples of `reshape`: in case of a matrix, it doesn't return a clone of the matrix, but mutates the matrix itself. The `reshape` method can do both (clone or mutate), but I think for consistency with the docs, array implementation, and other methods like `resize`, it should return a clone. I'll fix that. right away
